### PR TITLE
Add standalone wear support, fix BG sync to phone.

### PR DESCRIPTION
**Pull request by kskandispersonal**
Here is a new PR for the standalone wear support.  @jotomo helped me create a new branch in re-apply my changes.  I hope this looks better for merging:)  Thanks, and sorry for the time-consuming confusion!:(

### CODE REVIEW

**Comment by kskandispersonal** [comment]
@jamorham I just added some missing code from the server.  I will re-test this branch today after making the changes.

**Comment by kskandispersonal** [comment]
@jamorham While testing today, I noticed that the wear is getting slightly different BG results than that of xDrip+ on the phone.  I realize now why.  BgReading uses the latest 3 BgReadings to calculate the new BG, and currently wear does not save BGs from the phone.  I will need to look at this a bit to make a fix.  So please wait before you review the PR.  I will try to update you tomorrow.  In the mean time I am trying to keep this fork up to date at least as far as code conflicts on commits.

**Comment by jamorham** [comment]
Okay yes this is looking a lot better. I can see the changes a lot more clearly now. Please keep your //KS comments in there as they also really help.

This looks almost right except that I think even some changes were made while you were preparing it. I will annotate the section concerned that I noticed so far.

Thanks so much for doing this I think it will be a great addition. I really want to get it merged but as you know its pretty hefty.

File: app/src/main/java/com/eveningoutpost/dexdrip/BGHistory.java [hunk 1]
```diff
@@ -2,22 +2,16 @@
```
Comment (jamorham) [inline app/src/main/java/com/eveningoutpost/dexdrip/BGHistory.java]: I think the changes shown here are recent ones where your branch was out of sync. Can you checkout the current version of this and add that to nullify the changes to BGHistory.java

File: app/src/main/res/layout/activity_bghistory.xml [hunk 2]
```diff
@@ -33,23 +33,6 @@
                     android:layout_height="match_parent"
```
Comment (jamorham) [inline app/src/main/res/layout/activity_bghistory.xml]: This file also I don't think you intended changes to.

File: app/src/main/res/menu/menu_history.xml [hunk 3]
```diff
@@ -1,10 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
```
Comment (jamorham) [inline app/src/main/res/menu/menu_history.xml]: Also here.

File: build.gradle [hunk 4]
```diff
@@ -6,7 +6,7 @@ buildscript {
         mavenCentral()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.2.0'
```
Comment (jamorham) [inline build.gradle]: I think we need gradle 2.2.0 for Android studio 2.2

**Comment by jamorham** [comment]
Out of curiosity, how big does this make the microapk after compilation?

**Comment by kskandispersonal** [comment]
The debug apk is 3076KB.  I'm not sure what it was before I added code though.

**Comment by kskandispersonal** [comment]
I made the changes to save the minimum number of phone BGs (4 BGs I believe) used in BgReading calculation  on the wear device.  This seems to be working well and data is getting synced back to the phone correctly now.  I made a few additional changes related to processing the Wear Integration *connectG5 preferences.  I know once you start reviewing you'll find issues though!  

I am not really happy with the design.  Many of the Wear project files are coming directly from xDrip+ with only minor changes required.  It will make it hard to maintain wear since xDrip+/app is so actively being developed.  I thought about just using symbolic links to the app project directory source tree.  Hwr, the minor changes (between app and wear) typically include commenting out unneeded code on the wear side.  This reduces the wear build size and number of files that need to be included in the wear project.  Any suggestions on a better method?

Adrian mentioned he'd like a common Collector for this standalone feature.  I'm not sure how to implement that though since the wear integration is a project unto itself.  Maybe we can work on integrating the wear xDrip+ files back into the app project so there is a single code base instead of a separate code base for Wear?  Then once a single source is implemented, perhaps use symbolic links when building the wear apk or checking out src in git?

Other than issues you find, I think the standalone feature is largely complete.  It is still missing BG notifications though.  That will probably require re-using the xDrip+ existing alert/notification code as well.

**Comment by jotomo** [comment]
@kskandispersonal A few thoughts: the wear project needs to be a project since it's basically another app. The xdrip+ app depends on on the wear project (see app/build.gradle -> dependencies), since the common way is to install the app onto the phone which then syncs the 'embedded' wear app to the watch.
Symlinks are prone to break, also do they even work on Windows (outside of Cygwin/Mingw)?
I've only skimmed the code, but generally a shared library is a common approach to this: create a new project containing the code app and wear both use and have app and wear depend on that library. How doable this is depends on how loosely coupled the code to break out is.

**Comment by jamorham** [comment]
Symbolic links will probably create a big mess for windows users as I don't think git supports them properly from what I have read online.

My plan is to get this PR integrated, if there is the will to clean things up and produce shared classes later then we should definitely do that as it will make things better but I don't think that should be a barrier to getting this merged if it works and any negative impact is going to be minimal. 

The functionality is useful and important and @kskandispersonal has tested this and confirms it currently works.

The main problem with the PR is that there is something strange about its git history which means that every change made to the xdrip codebase results in this PR showing the changes in inverse. So I cannot simply merge it without risking undoing all other recent code changes.

I think the way to solve this is twofold. First @kskandispersonal can do whatever re-sync was done before to catch up to how things are today. For example I see that the note history feature recently shows up as an inverse change. Those differences need fixing again.

If we can do that then I will freeze changes to the code-base temporarily and then work to manually integrate the changes. There are 107 files so its not going to be fun but I think that many of those are new files which I can merge with zero impact and reduce the size of this PR to make it more manageable.  If the PR is then readable with the more difficult changes I can then attempt to merge those or do it by hand if we really have to.

I'm willing to do this but we will need to coordinate so that the time I have to freeze the code base is minimal. I will also need the resulting xDrip+ build tested as I do not have a G5.

**Comment by kskandispersonal** [comment]
Yes, I agree, a shared library would be a better approach.  I didn't want to make any major structural changes in xDrip+ though!

@jamorham I don't know why the note history is showing up as an inverse.  I'm not skilled at git so perhaps I am doing something wrong.  The "re-sync" you refer to is that the 'git reset upstream' that @jotomo helped me do:

`
$ git checkout -b standalone-wear 
$ git fetch upstream
$ git reset upstream/master 
$ git status 
$ rm log* 
$ git add . 
$ git commit -m "Add standalone wear support"
$ git push origin standalone-wear
`

**Comment by jamorham** [comment]
@kskandispersonal  Okay can you do that one more time and I will see what I can do with it then. I feel like it needs some kind of rebase or merge but I'm not quite sure exactly where the problem is coming from but I think if we can get it in sync then I can make some progress with it regardless. I'm not a git expert either and have had occasional problems with it so I don't want to introduce any other unknowns!

**Comment by kskandispersonal** [comment]
@jamorham Okay, I think I should create a new branch and reset to your upstream to get your latest code, then reapply my changes maybe through a git merge standalone-wear?   I'm not sure but I'll try it and see if that works.

**Comment by kskandispersonal** [comment]
@jotomo Would a rebase solve this issue?  Could I just rebase this branch to upstream?

`git rebase upstream/master`

**Comment by jamorham** [comment]
Superseded by https://github.com/jamorham/xDrip-plus/pull/14

**Pull request closed without merging**